auth.go

```diff
@@ -40,6 +42,7 @@ var (
 	ErrInvalidSessionToken    = errors.New("Invalid session token")
 	ErrInvalidPasswordToken   = errors.New("Invalid password token")
 	ErrPasswordTokenExpired   = errors.New("Password token has expired")
+	ldapUsed                  = false
```

Comment:
A global variable like this (ldapUsed) doesn't belong here. Classic scope leakage.